oops/oops.go

```diff
@@ -130,6 +130,34 @@ func Frames(err error) [][]Frame {
 	return parsedStacks
 }
 
+// DropStackFrames drops numFrames from the stack trace and returns a new copy of the error
+func DropStackFrames(err error, numFrames uint) error {
```

Comment:
nit: can we call it `SkipFrames` to match the golang terminology? https://golang.org/pkg/runtime/#Caller